contracts/auto-withdrawer/src/contract.rs

```diff
@@ -106,106 +116,114 @@ fn execute_bond_with_ld_assets(
+    let existing_bonding = bondings_map().may_load(deps.storage, &bonding_id)?;
+    if let Some(existing_bonding) = existing_bonding {
+        deposit = merge_coin_vecs(existing_bonding.deposit, deposit);
+        withdrawal_asset.amount += existing_bonding.withdrawal_amount;
+    }
     bondings_map().save(
         deps.storage,
-        &token_id,
+        &bonding_id,
         &BondingRecord {
             bonder: info.sender,
+            withdrawal_amount: withdrawal_asset.amount,
             deposit,
         },
     )?;
 
-    let withdrawal_voucher = WITHDRAWAL_VOUCHER_ADDRESS.load(deps.storage)?;
-    let msg = WasmMsg::Execute {
-        contract_addr: withdrawal_voucher.into_string(),
-        msg: to_json_binary(
-            &drop_staking_base::msg::withdrawal_voucher::ExecuteMsg::TransferNft {
-                recipient: env.contract.address.into_string(),
-                token_id,
-            },
-        )?,
-        funds: vec![],
-    };
-
     // TODO: attributes
-    Ok(Response::new().add_message(msg))
+    Ok(Response::new())
 }
 
 fn execute_unbond(
     deps: DepsMut<NeutronQuery>,
     info: MessageInfo,
-    token_id: String,
+    batch_id: Uint128,
 ) -> ContractResult<Response<NeutronMsg>> {
-    let bonding = bondings_map().load(deps.storage, &token_id)?;
+    let bonding_id = get_bonding_id(&info.sender, batch_id);
+    let bonding = bondings_map().load(deps.storage, &bonding_id)?;
     ensure_eq!(info.sender, bonding.bonder, ContractError::Unauthorized {});
-    bondings_map().remove(deps.storage, &token_id)?;
-
-    let withdrawal_voucher = WITHDRAWAL_VOUCHER_ADDRESS.load(deps.storage)?;
-
-    let nft_msg: CosmosMsg<NeutronMsg> = WasmMsg::Execute {
-        contract_addr: withdrawal_voucher.into_string(),
-        msg: to_json_binary(
-            &drop_staking_base::msg::withdrawal_voucher::ExecuteMsg::TransferNft {
-                recipient: info.sender.to_string(),
-                token_id,
-            },
-        )?,
-        funds: vec![],
-    }
-    .into();
-
-    let deposit_msg = BankMsg::Send {
-        to_address: info.sender.into_string(),
-        amount: bonding.deposit,
-    }
-    .into();
+    bondings_map().remove(deps.storage, &bonding_id)?;
+
+    let withdrawal_denom_prefix = WITHDRAWAL_DENOM_PREFIX.load(deps.storage)?;
+    let withdrawal_token_address = WITHDRAWAL_TOKEN_ADDRESS.load(deps.storage)?;
+    let withdrawal_denom =
+        get_full_withdrawal_denom(withdrawal_denom_prefix, withdrawal_token_address, batch_id);
+
+    let send_assets_msg: BankMsg = BankMsg::Send {
+        to_address: info.sender.to_string(),
+        amount: merge_coin_vecs(
+            vec![Coin::new(
+                bonding.withdrawal_amount.u128(),
+                withdrawal_denom,
+            )],
+            bonding.deposit,
+        ),
+    };
 
     // TODO: attributes
-    Ok(Response::new().add_messages([nft_msg, deposit_msg]))
+    Ok(Response::new().add_message(send_assets_msg))
```

Comment:
here and then, do you mind replacing `Response:new` with `response` helper with some informative attrs?